packages/client/components/promptResponse/PromptResponseEditor.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {useEffect, useState} from 'react'
+import Placeholder from '@tiptap/extension-placeholder'
+import {EditorContent, EditorEvents, useEditor, JSONContent} from '@tiptap/react'
+import StarterKit from '@tiptap/starter-kit'
+
+interface Props {
+  autoFocus?: boolean
+  initialDoc: JSONContent
+  handleSubmit: (value: string) => void
+  readOnly: boolean
+  placeholder: string
+}
+
+const PromptResponseEditor = (props: Props) => {
+  const {autoFocus: autoFocusProp, initialDoc,  handleSubmit, readOnly, placeholder} = props
+  const [isEditing, setIsEditing] = useState(false)
+  const [autoFocus, setAutoFocus] = useState(autoFocusProp)
+  const [doc, setDoc] = useState(initialDoc)
+  useEffect(() => {
+    if (isEditing) return
+    setDoc(initialDoc)
+  }, [initialDoc])
+
+  const setEditing = (isEditing: boolean) => {
+    setIsEditing(isEditing)
+    setAutoFocus(false)
+  }
+
+  const onUpdate = ({editor}: EditorEvents['update']) => {
```

Comment:
-1 this looks wrong. we shouldn't be serializing the entire document on update.
the only time we should serialize is when we send it to the server.